x-pack/plugins/cases/server/attachment_framework/so_references.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectReference } from '@kbn/core/types';
+import { CommentRequest, CommentRequestPersistableStateType } from '../../common/api';
+import { isCommentRequestTypePersistableState } from '../common/utils';
+import { PersistableStateAttachmentTypeRegistry } from './persistable_state_registry';
+
+interface SavedObjectAttributesAndReferences {
+  state: CommentRequestPersistableStateType;
+  references: SavedObjectReference[];
+}
+
+interface ExtractDeps {
+  persistableStateAttachmentTypeRegistry: PersistableStateAttachmentTypeRegistry;
+}
+
+type InjectDeps = ExtractDeps;
+
+export function extractPersistableStateReferences(
+  state: CommentRequestPersistableStateType,
+  deps: ExtractDeps
+): SavedObjectAttributesAndReferences {
+  const { persistableStateAttachmentState, persistableStateAttachmentTypeId } = state;
+
+  if (!deps.persistableStateAttachmentTypeRegistry.has(persistableStateAttachmentTypeId)) {
+    return { state, references: [] };
+  }
+
+  const attachment = deps.persistableStateAttachmentTypeRegistry.get(
+    persistableStateAttachmentTypeId
+  );
+
+  const { state: persistableState, references: persistableStateReferences } = attachment.extract({
+    persistableStateAttachmentState,
+    persistableStateAttachmentTypeId,
+  });
+
+  return {
+    state: { ...state, ...persistableState, persistableStateAttachmentTypeId },
```

Comment:
I put the old `persistableStateAttachmentTypeId` here in case someone changes it by mistake in the `extract` function